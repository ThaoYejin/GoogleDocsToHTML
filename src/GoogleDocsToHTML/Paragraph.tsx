import React from "react";
import { docs_v1 } from "googleapis";
import { TextRun } from "./TextRun";
import { Image } from "./Image";


export interface PTagProps {
    paraObj?: docs_v1.Schema$Paragraph;
    doc: docs_v1.Schema$Document;
}

export const Paragraph: React.FC<PTagProps> = ({
    paraObj,
    doc
}) => {
    var content: React.ReactNode[];
    content = [];
    var pObjElement = paraObj?.elements;
    pObjElement?.forEach(function (element, index) {
        if (element.inlineObjectElement != undefined) {
            content?.push(
                <Image key={index} doc={doc} element={element} ></Image>
            )
        } else if (element.inlineObjectElement == undefined) {
            content?.push(
                <p key={index}>
                    <TextRun element={element}></TextRun>
                </p>
            )
        }
    })
    return (
        <>
            {content}
        </>
    );
}