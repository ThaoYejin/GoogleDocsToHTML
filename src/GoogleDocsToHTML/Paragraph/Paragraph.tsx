import React from "react";
import { docs_v1 } from "googleapis";
import { TextRun } from "../TextRun/TextRun";
import { Image } from "../Image/Image";
import { Link } from "../Link/Link";


export interface PTagProps {
    paraObj?: docs_v1.Schema$Paragraph;
    doc: docs_v1.Schema$Document;
}

export const Paragraph: React.FC<PTagProps> = ({
    paraObj,
    doc
}) => {
    var content: React.ReactNode[];
    content = [];
    var pObjElement = paraObj?.elements;
    pObjElement?.forEach(function (element, index) {
        if (element.inlineObjectElement != undefined) {
            content?.push(
                <Image key={index} doc={doc} element={element} ></Image>
            )
        }
        if (element.textRun?.textStyle?.link != undefined) {
            content.push(
                <Link paragraphElement={element}></Link>
            )
        } 
        if (element.inlineObjectElement == undefined && element.textRun?.textStyle?.link == undefined) {
            content?.push(
                <TextRun key={index} element={element}></TextRun>
            )
        }
    })
    return (
        <p>
            {content}
        </p>
    );
}