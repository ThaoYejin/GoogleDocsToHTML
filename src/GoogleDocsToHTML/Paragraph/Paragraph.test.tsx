import React from 'react';
import renderer from 'react-test-renderer';
import { Paragraph } from "./Paragraph";
import * as docImage from './ParagraphImage.json';
import * as docLink from './ParagraphLink.json';
import * as docTextRun from './ParagraphTextRun.json';

it("Paragraph Image runs correctly", ()=>{
    var paraObj = {
        "elements": [
          {
            "startIndex": 1,
            "endIndex": 2,
            "inlineObjectElement": {
              "inlineObjectId": "kix.fw30sug1ofhp",
              "textStyle": {
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          },
          {
            "startIndex": 2,
            "endIndex": 3,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          }
        }
      };
      const tree = renderer
        .create(<Paragraph paraObj={paraObj} doc={docImage} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

it("Paragraph Link runs correctly", ()=>{
    var paraObj = {
        "elements": [
          {
            "startIndex": 1,
            "endIndex": 8,
            "textRun": {
              "content": "Thao FB",
              "textStyle": {
                "underline": true,
                "foregroundColor": {
                  "color": {
                    "rgbColor": {
                      "red": 0.06666667,
                      "green": 0.33333334,
                      "blue": 0.8
                    }
                  }
                },
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                },
                "link": {
                  "url": "https://www.facebook.com/thaoho0806"
                }
              }
            }
          },
          {
            "startIndex": 8,
            "endIndex": 9,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          }
        }
      };
      const tree = renderer
        .create(<Paragraph paraObj={paraObj} doc={docLink} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

it("Paragraph TextRun runs correctly", ()=>{
    var paraObj = {
        "elements": [
          {
            "startIndex": 1,
            "endIndex": 31,
            "textRun": {
              "content": "This is a normal paragraph!!!!",
              "textStyle": {
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          },
          {
            "startIndex": 31,
            "endIndex": 32,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          }
        }
      };
      const tree = renderer
        .create(<Paragraph paraObj={paraObj} doc={docTextRun} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})