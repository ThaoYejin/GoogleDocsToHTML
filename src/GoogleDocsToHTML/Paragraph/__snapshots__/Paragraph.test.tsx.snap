// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Image runs correctly 1`] = `
<p>
  <img
    src="https://lh5.googleusercontent.com/227wZYprRh_IEImeCfwbwrTFEPCnGhADmEmc9m91RnrHwut-a20xv5Ls0ApQRj8lqG5MRwXkRDulj4ddTczJPnIOzVeIaOGfoywS9By-nJc5W59VW4M-UEwx0f1cwhOfB39asOg_2D4V7Jdhfaocthxa1n8"
  />
  <span
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    

  </span>
</p>
`;

exports[`Paragraph Link runs correctly 1`] = `
<p>
  <a
    href="https://www.facebook.com/thaoho0806"
  >
    <span
      style={
        Object {
          "color": "rgb(17.00000085,85.0000017,204)",
          "fontSize": 12,
          "textDecoration": "underline",
        }
      }
    >
      Thao FB
    </span>
  </a>
  <span
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    

  </span>
</p>
`;

exports[`Paragraph TextRun runs correctly 1`] = `
<p>
  <span
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    This is a normal paragraph!!!!
  </span>
  <span
    style={
      Object {
        "fontSize": 12,
      }
    }
  >
    

  </span>
</p>
`;
